dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -66,265 +69,189 @@ public static void prepare() throws Exception {
-            int action2 = 301;
-            int order2 = 1;
-
-            contentlet2 = filtersUtil.createVanityUrl(title2, site2, uri2,
-                    forwardTo2, action2, order2, defaultLanguageId);
-
-            CachedVanityUrl vanityURLCached = vanityUrlCache.get(new CacheVanityKey(
-                                                                        contentlet1.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                                                                        contentlet1.getLanguageId(),
-                                                                        contentlet1.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                                                                    ));
-            Assert.assertNull(vanityURLCached);
 
-            CachedVanityUrl vanity = vanityUrlAPI.getLiveCachedVanityUrl(uri, defaultHost, defaultLanguageId, user);
 
-            Assert.assertNotNull(vanity);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity.getVanityUrlId());
-            Assert.assertEquals(site, vanity.getSiteId());
-            Assert.assertEquals(uri, vanity.getUrl());
-            Assert.assertEquals(forwardTo, vanity.getForwardTo());
-            Assert.assertEquals(action, vanity.getResponse());
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
+        String site = defaultHost.getIdentifier();
+        String uri = "/test1_" + i;
+        String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
+        int action = 200;
+        int order = 1;
 
-            vanityURLCached = vanityUrlCache.get(new CacheVanityKey(
-                    contentlet1.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                    contentlet1.getLanguageId(),
-                    contentlet1.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-            ));
-            Assert.assertNotNull(vanityURLCached);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanityURLCached.getVanityUrlId());
+        contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
 
-            vanityURLCached = vanityUrlCache.get(new CacheVanityKey(
-                    contentlet2.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                    contentlet2.getLanguageId(),
-                    contentlet2.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-            ));
-            Assert.assertNull(vanityURLCached);
+        
+        
+        Optional<CachedVanityUrl> optionalCache = cache.getDirectMapping(uri, defaultHost, defaultLanguage);
+        assert(optionalCache==null );
+        
 
-            vanity = vanityUrlAPI.getLiveCachedVanityUrl(uri2, defaultHost, defaultLanguageId, user);
+        Optional<CachedVanityUrl> vanityCached = vanityUrlAPI.resolveVanityUrl(uri, defaultHost, defaultLanguage);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'vanityCached' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)